client/server visualizer

edofic opened the pull request:
bugfix za classloader

MainWithPublish - Main z serverjem za vizualizacijo
CliServer - server brez gui, podpira kliente za vizualizacijo
CliVisulizer - povezava preko socketa na server in prikaz z TerminalView
GuiVIsulizer - predelan Main, ki prek socketa spremlja igro na serverju

protokol za pošiljanje igre čez socket je stateless in precej primitiven. na localhost testirano dela tekoče tudi za največji map(80x80). če bo potreba čez network in bo premalo bandwidtha lahko tudi kaj optimiziram.

lukacu commented:
enkratno, bom probal cimprej stestirati, pol pa mergam. drugace pa fajn, da v commitih uporabljas anglescino, dajmo jo se v pull requestih, ker nikoli ne vemo, kdaj postanemo mednarodni :)

lukacu commented:
What is the exact point of MainWithPublish?

edofic commented:
I added state publishing(for remote visualization). I thought that would mean messing with code a lot so I made a copy of the class first, so I can run Main too. It turned out it's just few lines of code. So I "merged" changes into single class. Commit 612b131.

My IDE did some automatic reformatting and screwed up diffs. In fact I just added lines 55 and 366.
